### DISCUSSION TITLE
Did anyone manage to export a project using this extension in a CI/CD pipeline?

### DISCUSSION THREAD

**Discussion by 3ter** [utopia-rise/fmod-gdextension#221] [Q&A]
We're using https://github.com/abarichello/godot-ci which is using a Ubuntu image as base for exporting and deploying the Godot project.

Unfortunately it seems that using the extension on Linux is hard. I've checked all the libraries with `ldd` and everything checked out. But still it tries to open the `.dll` files instead of the `.so` files for Linux.

Any help would be appreciated as it's a bit sad to kiss our pipelines goodbye for this extension.

**Comment by plumocracy** [comment]
I'm working on this problem today. My idea is spinning up a docker container to build/compile in so it can be done on windows instead of linux. I'll report back if I'm successful with the steps/workflow I use.

**Reply by 3ter** [reply]
So do you plan to replace the image in the Dockerfile (e.g. https://github.com/abarichello/godot-ci/blob/bd9eea16d14bf71f5876b71b16955218df36266b/Dockerfile#L1) with something like https://github.com/dockur/windows? Thanks for experimenting with this, I appreciate it 😊!

**Reply by plumocracy** [reply]
I'm probably just going to spin up a new docker container for our games ci/cd running windows cli (the link you provided I tried today and it takes newly an hour to download and install a GUI version of windows which we don't really need for this scenario) and try to build the game there. Automating this is important to us bc we're a small team with limited time/resources so I'll figure it out eventually it might just take a week or so. Maybe I'll figure out a Linux based solution in docker who knows. I'll let you know for sure tho once I have reproducible steps!

**Comment by meloncolle** [comment]
I know this is a pretty old discussion, but I came across it while trying to figure out build automation with this plugin. I thought I'd share my solution in case anyone else comes across this! [Here is my github workflow file](https://github.com/meloncolle/godot-demo-projects/blob/master/.github/workflows/build.yml).

One other thing I had to do was [update the plugin's .gitignore](https://github.com/meloncolle/godot-demo-projects/commit/ed1ebd64bd042285a7b0a95917af10d217d89c74) and then reinstall the plugin. Seems like it was mistakenly excluding some libraries, which causes problems if the repository is cloned.
